plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/BooleanCoercer.java

```diff
@@ -52,4 +55,65 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
             toType.writeSlice(blockBuilder, converted);
         }
     }
+
+    public static TypeCoercer<VarcharType, BooleanType> createVarcharToBooleanCoercer(VarcharType fromType, boolean isOrcFile)
+    {
+        return isOrcFile ? new OrcVarcharToBooleanCoercer(fromType) : new VarcharToBooleanCoercer(fromType);
+    }
+
+    public static class VarcharToBooleanCoercer
+            extends TypeCoercer<VarcharType, BooleanType>
+    {
+        // These values are populated from
+        // https://github.com/apache/hive/blob/4df4d75bf1e16fe0af75aad0b4179c34c07fc975/serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/PrimitiveObjectInspectorUtils.java#L559
```

Comment:
Should mark as it is compatible with Hive 3.0. See https://issues.apache.org/jira/browse/HIVE-15939